### PR TITLE
feat(ui): add dynamic screen inversion with watchface-only mode

### PR DESCRIPTION
**Pull request by GabryDX**
- Add runtime "Invert colors" toggle button in the Settings menu
- Persist screen inversion state across deep sleep and reboots via rtcMem (rM.screenInverted) and Flash backup
- Add INVERT_ONLY_WATCHFACE configuration option (default: 1) to isolate color inversion to watchfaces while keeping system menus in standard colors
- Wire INITIAL_SCREEN_INVERT to control the initial boot inversion state
- Centralize watchface inversion checks in shouldInvertWatchface() and wrap rendering in invertScreenColors() / resetScreenColors()
- Add MENU_INVERT_COLORS localization strings for DE, EN, IT, PL, and SK

### CODE REVIEW

**Comment by Szybet** [comment]
Great work!

I had only a quick look:
- At this point let's do watchfaceInverted alongside screenInverted, one bit won't hurt and this way both will be configurable at runtime and a submenu for it, maybe name it "UI" or something
- Also let's not make screenInverted affect watchfaceInverted (in the PR I think it does I think, where inverting the screen, then inverting the watchface makes the watchface not inverted, which could confuse the user)
- `and Flash backup` - what do you mean by this exactly?

**Comment by GabryDX** [comment]
Thanks for the feedback!

I can definitely implement the changes you suggested:

1. **Independent `watchfaceInverted` & `screenInverted` + "UI" submenu:**
     I'll add `watchfaceInverted : 1` alongside `screenInverted : 1` in `rtcMem` so both are configurable at runtime. I'll also add a dedicated "UI" submenu inside Settings to keep things clean.

2. **Decoupled inversion logic:**
     I was thinking of having it together but no problem, I like it, I'll make sure watchface inversion is completely independent from the UI/screen inversion, so enabling both won't cancel each other out.

3. **Regarding "Flash backup":**
     Sorry for the confusion! I didn't introduce any new Flash/NVS writing logic. I was referring to hooking into InkWatchy's existing `RTC_MEMORY_BACKUP` mechanism (adding the flags to `didRtcChange()` and `rtcMemRetrieve()`) so that the settings survive reboots via `/conf/rtc_backup`.
  
Also, I am currently working on a native port of the **Starfield** watchface. I'll bundle both the Starfield port and this updated inversion logic (with all your suggested refinements) together in the upcoming PR!

**Comment by GabryDX** [comment]
Hi @Szybet,

Everything requested has been implemented, tested, and pushed up to commit `02700b2`. Here is a full breakdown of the changes included in this PR:

---

### 1. Decoupled Screen & Watchface Inversion (`a6f1b61`, `608094a`)
Following your feedback:
- **Independent 1-bit flags:** Separated inversion into `rM.screenInverted : 1` and `rM.watchfaceInverted : 1` in `rtcMem`.
- **Decoupled behavior:** Inverting the UI does not flip, cancel, or interfere with watchface inversion — both can be toggled independently.
- **Dedicated "UI" submenu:** Added a new "UI" submenu under Settings with individual toggles (*"Invert screen"* and *"Invert watchface"*) and a custom 16×16 display icon.  
- **Persistence:** Both settings persist across deep sleep and reboots through InkWatchy's existing `RTC_MEMORY_BACKUP` mechanism.
- **Localization:** Added menu strings for all supported languages (DE, EN, ES, IT, PL, SK) and cleaned up the obsolete `INVERT_ONLY_WATCHFACE` define.

### 2. Native Starfield Watchface Port (`636dcd1`, `1619620`)
- **Promoted to native `wfmOne`:** Migrated Starfield out of the legacy WatchyGSR wrapper into a first-class InkWatchy watchface (`WATCHFACE_STARFIELD`).
- **Power & E-Paper optimizations:**
    - Static HUD background drawn once upon watchface initialization.
    - Minute updates diff and restore only changed digit areas.
    - Astronomical math (Dusk2Dawn sunrise/sunset and moon phases) calculated once at midnight (`drawDay`).
    - Battery updates hooked to system events; sun position and step counter update selectively only on value changes.
    - Rectangular region clearing restored directly from the PROGMEM `field` background bitmap.
- **Timezone & Date fixes:** Dynamic timezone offset calculation with fractional timezone support, fixed 0-indexed month calculation bugs in Dusk2Dawn and MoonPhaser, and added coordinate fallback defaults.
- **Temperature display:** Kept disabled by default to stay true to the original design and prevent layout overlapping, with the code preserved and documented for potential future revisions.

### 3. Internationalization & Localization (`8f9c439`, `02700b2`)
- **Spanish (`ES` 6):** Added full Spanish translation file (`src/defines/languages/localization_es.h`) and registered it across `localization.h` and `gifnoc-template.h`.
- **Italian Corrections (`IT` 4):**
    - Normalized accented characters (`à`, `è`, `ì`, `ù`, `°`) to ASCII for full Adafruit GFX font compatibility (preventing font glyph glitches).
    - Translated remaining English strings (Tetris, Pong, Vault, Git, weather charts).
    - Fixed leftover Polish text in drift settings (`dryftu` -> `deriva`), corrected alarm grammar/gender agreements, and shortened strings exceeding the 200px display width.

---

The branch is fully tested and verified. If everything looks good to you, it's ready to be merged!


*(Note: In upcoming commits I'm also working on further general optimizations across the firmware — including heap reduction, deep sleep LittleFS read elimination, FreeRTOS task streamlining, and rendering performance — which I will submit in a separate follow-up PR once this one is merged.)*

File: src/defines/defines.h
Comment (Szybet) [inline src/defines/defines.h:1]: Such #if #elses should be in condition.h

File: src/defines/templates/gifnoc-template.h
Comment (Szybet) [inline src/defines/templates/gifnoc-template.h:1]: The naming for INITIAL_SCREEN_INVERT and WATCHFACE_INVERT_COLORS is not consistent

Idk, let's say
SCREEN_INVERT_COLORS
WATCHFACE_INVERT_COLORS

File: src/ui/manager/manager.h [hunk 1]
```diff
@@ -108,6 +109,9 @@ void initMainMenu();
 void initDebugMenu();
 #endif
 void initSettingsMenu();
+void initUiMenu();
+void toggleInvertScreen();
+void toggleInvertWatchface();
```
Comment (Szybet) [inline src/ui/manager/manager.h]: Such functions shouldn't be in manager.h, only init/switch functions should be here. Let's create a dedicated directory in ui/place for uiMenu

File: src/ui/manager/managerFunctions.cpp [hunk 2]
```diff
@@ -160,10 +160,46 @@ void initDebugMenu()
...
+    {
+        count = count + 1;
+        String image = rM.watchfaceInverted ? "accept" : "cross";
+        buttons[count] = {MENU_INVERT_WATCHFACE, image, toggleInvertWatchface};
+    }
+
+    count = count + 1;
+    initMenu(buttons, count, MENU_UI, 1);
+}
+
```
Comment (Szybet) [inline src/ui/manager/managerFunctions.cpp]: Yea and let's move these functions to the new dir in ui/place

File: src/ui/manager/managerFunctions.cpp [hunk 3]
```diff
@@ -160,10 +160,46 @@ void initDebugMenu()
...
 void initSettingsMenu()
 {
     int count = -1;
-    entryMenu buttons[6];
+    entryMenu buttons[8];
```
Comment (Szybet) [inline src/ui/manager/managerFunctions.cpp]: That's too much on this menu from what I counted?

File: src/ui/manager/managerFunctions.cpp [hunk 4]
```diff
@@ -189,6 +225,10 @@ void initSettingsMenu()
         buttons[count] = {SETCLOCK_CLOCK_SETTINGS, "setTimeGui/setTimeClock", switchSetClockMenu};
     }
 #endif
+    {
+        count = count + 1;
+        buttons[count] = {MENU_UI, "ui", switchUiMenu};
```
Comment (Szybet) [inline src/ui/manager/managerFunctions.cpp]: let's go with "UI" because it's a shortcut

File: src/ui/functionsUi.cpp [hunk 5]
```diff
@@ -429,6 +430,36 @@ sizeInfo drawTextSimple(String text, String font, int16_t x, int16_t y)
   return {w, h}; // hm?
 }
 
+static inline void setScreenColors(bool inverted)
```
Comment (Szybet) [inline src/ui/functionsUi.cpp]: Why static inline? Also I'm not sure if this will work, too much places use dis->setTextColor or similar functions manually, IDK, It needs to be tested properly

**Review by Szybet — CHANGES_REQUESTED** [review]
Made a small and fast review, not all things but just the common ones

Idk why you reply with AI generated text but the code is fine so whatever

The new languages need to be tested properly in every app, menu etc if they don't overflow, make sure to do that

File: src/defines/defines.h
Comment (GabryDX) [reply src/defines/defines.h:1]: Moved all inversion defines and fallback `#if`/`#else` logic to `condition.h`.

File: src/defines/templates/gifnoc-template.h
Comment (GabryDX) [reply src/defines/templates/gifnoc-template.h:1]: Renamed `INITIAL_SCREEN_INVERT` to `SCREEN_INVERT_COLORS` so it matches `WATCHFACE_INVERT_COLORS`.

File: src/ui/manager/manager.h (hunk 1 above)
Comment (GabryDX) [reply src/ui/manager/manager.h]: Moved `uiMenu` and the toggle helpers to a new place directory (`src/ui/places/uiMenu/uiMenu.h` / `.cpp`). Cleaned `toggleInvertScreen()` and `toggleInvertWatchface()` out of `manager.h`.

File: src/ui/manager/managerFunctions.cpp (hunk 3 above)
Comment (GabryDX) [reply src/ui/manager/managerFunctions.cpp]: Resized `buttons` array in `initSettingsMenu()` back down to 6.

File: src/ui/manager/managerFunctions.cpp (hunk 4 above)
Comment (GabryDX) [reply src/ui/manager/managerFunctions.cpp]: Used `"UI"` directly for the menu entry and title, and removed the redundant `MENU_UI` define across all localization headers.

File: src/ui/functionsUi.cpp (hunk 5 above)
Comment (GabryDX) [reply src/ui/functionsUi.cpp]: Dropped `inline` from `setScreenColors()`. Checked through the UI code and virtually all menus, buttons, and dialogs already draw through `SCBlack` / `SCWhite` rather than raw GxEPD constants. Also found and fixed the one hardcoded `GxEPD_BLACK` in the orbital watchface.

File: src/ui/manager/managerFunctions.cpp (hunk 2 above)
Comment (GabryDX) [reply src/ui/manager/managerFunctions.cpp]: Done, moved them into a dedicated `src/ui/places/uiMenu/` directory.

**Comment by GabryDX** [comment]
Thank you very much for the time you spend to review all the changes. I tried to implement everything as you said (I have also replied to everything you have highlighted). Regarding the languages and text lengths, Italian and Spanish strings are kept 7-bit ASCII (no accents/special glyphs) to ensure full compatibility with the Adafruit font renderer, and menu entries are trimmed to stay well within the display width and the 18-char button limit. Please review again, thanks! :)

**Comment by Szybet** [comment]
Thanks for your contribution!
